e2e/test/scenarios/admin/admin-reproductions.cy.spec.js

```diff
@@ -195,8 +195,8 @@ describe("(metabase#46714)", () => {
       cy.findByText("Orders").click();
     });
 
-    //TODO: Fix this shame
-    cy.wait(2000);
+    cy.findByTestId("entity-picker-modal").should("not.exist");
+    cy.findByTestId("segment-editor").findByText("Orders").should("exist");
```

Comment:
```suggestion
    cy.findByTestId("segment-editor").findByText("Orders").should("be.visible");
```

generally testing for visibility rather than existence is better practice. kinda doesn't matter if elements exist if you can't see them